packages/flutter_test/lib/src/controller.dart

```diff
@@ -23,6 +23,144 @@ const double kDragSlopDefault = 20.0;
+    SemanticsAction.scrollLeft.index |
+    SemanticsAction.scrollRight.index;
+
+  static final int _importantFlags =
+    SemanticsFlag.hasCheckedState.index |
+    SemanticsFlag.hasToggledState.index |
+    SemanticsFlag.hasEnabledState.index |
+    SemanticsFlag.isButton.index |
+    SemanticsFlag.isTextField.index |
+    SemanticsFlag.isFocusable.index |
+    SemanticsFlag.isSlider.index |
+    SemanticsFlag.isInMutuallyExclusiveGroup.index;
+
+
+  /// A reference to the current instance of the binding.
+  final WidgetsBinding binding;
+
+  /// Attempts to find the [SemanticsNode] of first result from `finder`.
+  ///
+  /// If the object identified by the finder doesn't own it's semantic node,
+  /// this will return the semantics data of the first ancestor with semantics.
+  /// The ancestor's semantic data will include the child's as well as
+  /// other nodes that have been merged together.
+  ///
+  /// If the [SemanticsNode] of the object identified by the finder is
+  /// force-merged into an ancestor (e.g. via the [MergeSemantics] widget)
+  /// the node into which it is merged is returned. That node will include
+  /// all the semantics information of the nodes merged into it.
+  ///
+  /// Will throw a [StateError] if the finder returns more than one element or
+  /// if no semantics are found or are not enabled.
+  SemanticsNode get(Finder finder) {
+    if (binding.pipelineOwner.semanticsOwner == null) {
+      throw StateError('Semantics are not enabled.');
+    }
+    final Iterable<Element> candidates = finder.evaluate();
+    if (candidates.isEmpty) {
+      throw StateError('Finder returned no matching elements.');
+    }
+    if (candidates.length > 1) {
+      throw StateError('Finder returned more than one element.');
+    }
+    final Element element = candidates.single;
+    RenderObject? renderObject = element.findRenderObject();
+    SemanticsNode? result = renderObject?.debugSemantics;
+    while (renderObject != null && (result == null || result.isMergedIntoParent)) {
+      renderObject = renderObject.parent as RenderObject?;
+      result = renderObject?.debugSemantics;
+    }
+    if (result == null) {
+      throw StateError('No Semantics data found.');
+    }
+    return result;
+  }
+
+  /// Simulates a traversal of the currently visible semantics tree.
+  ///
+  /// Starts at the node for `start`. If `start` is not provided, then returns
+  /// a traversal of all currently available nodes.
+  ///
+  /// Since the order is simulated, edge cases that differ between platforms
+  /// (such as how the last item in a scrollable list is handled) may be
+  /// inconsistent with platform behavior, but are expected to be sufficient for
+  /// testing order, availability to accessibility services, and interactions.
+  List<SemanticsNode> simulatedTraversal({Finder? start}) {
+    if (binding.pipelineOwner.semanticsOwner == null) {
+      throw StateError('Semantics are not enabled.');
+    }
```

Comment:
I ended up moving this out to the `WidgetController.semantics` getter. Since _anything_ in this class is going to require semantics to be enabled, I figured we may as well do the check early and enforce it for _any_ access of that property.